board.py

```diff
@@ -6,16 +6,37 @@ def get_board(self):
         return self._board
 
     def get_board_range(self):
-        board_range = range(1, len(self._board) * len(self._board))
+        board_range = range(1, len(self._board) * len(self._board) + 1)
         return board_range
 
+    def get_board_winner_status(self):
+        board = self._board
+
+        for row in board:
+            if all(elem == row[0] for elem in row):
+                return True
+
+        for col in range(len(board)):
+            if len({row[col] for row in board}) == 1:
+                return True
+
+        if len({board[i][i] for i in range(len(board))}) == 1:
+            return True
+
+        if len({board[i][len(board) - i - 1] for i in range(len(board))}) == 1:
+            return True
+
+        if all(not isinstance(cell, int) for row in board for cell in row):
+            return None
+
+        return False
+
     def update_board(self, player, cell_number):
         for row_idx, row in enumerate(self._board):
             for col_idx, cell in enumerate(row):
                 if cell == cell_number:
                     self._board[row_idx][col_idx] = player
 
-
     def __str__(self):
         board_layout = "************************\n"\
```

Comment:
When you describe the responsibilities of a Board class, does it feel like formatting is consistent with its other responsibilities?

Reply:
no and all of this is definitely getting moved to its own class on the next ticket that deals with board formatting... because that code is gonna be pretty extensive I think.

Reply:
just to follow up on this. do you think board formatting refactors/movings should happen on this ticket or should I wait until the following tickets where board formatting will be dramatically reworked and moved?

Reply:
that's fine with me, forgot that you had that ticket up on deck next 👍